crates/wysiwyg/src/dom/nodes/text_node.rs

```diff
@@ -60,17 +61,71 @@ where
     pub fn set_handle(&mut self, handle: DomHandle) {
         self.handle = handle;
     }
+
+    fn handle_several_whitespaces(
+        formatter: &mut HtmlFormatter<S>,
+        pos: usize,
+        is_last_node_in_parent: bool,
+    ) {
+        let mut ranges_to_replace = Vec::new();
+        let mut current_range = usize::MAX..usize::MAX;
+        let mut whitespaces: usize = 0;
+        let mut needs_to_replace = false;
+        let w = S::c_from_char(' ');
+        let nbsp = S::c_from_char('\u{A0}');
```

Comment:
Hey @jmartinesp, I'm currently working on a refactoring of the `UnicodeString` trait and during that, noticed that this is a problematic use of `c_from_char`. It will panic if `S` is `String` because with `CodeUnit = u8`, the nbsp character doesn't fit inside a single code unit.

Reply:
I think the right fix would be to change the inner buffer of `HtmlFormatter` to be `S` instead of `Vec<S::CodeUnit>`, and update `chars_from` to iterate over `char`s instead of code units here. Let me know whether you think that is a good idea. I'm also reachable on Matrix (@jplatte:flipdot.org).

Reply:
Actually, the `let mut whitespaces: usize = 0;` used here kind of works around either this issue or a similar one since I remember facing panics for utf8 strings, but it's probably better to re-write HtmlFormatter. 

@andybalaam since you wrote most of it and the abstractions around strings, do you think it's worth it?

Reply:
I don't have full context, but I think using `S` as the buffer of `HtmlFormatter` sounds right.